### PR TITLE
Support FP8 in batch split config

### PR DESCRIPTION
**Pull request by BirdsOfAFthr**
# Description

This update enables FP8 quantization support for DeepSeek batch split configurations.

When quantization is active, the following changes apply:

* Kernel Quantization: gmm kernels allow FP8 recipes (defined via the MaxText command line) in both forward and backward passes.

* `gmm` Weight All-Gathers: gmm weight all-gathers are now quantized along non-expert, non-tensor, and non-tensor_transpose axes.

Note: Quantization for dispatch all-gathers is currently out of scope and will be implemented in a follow-up PR.

# Tests

* Verification: Validated via end-to-end (e2e) perf and convergence benchmarks.

* Coverage: Unit tests will be added in a subsequent update.

# Checklist

Before submitting this PR, please make sure (put X in square brackets):
- [X] I have performed a self-review of my code. For an optional AI review, add the `gemini-review` label.
- [X] I have necessary comments in my code, particularly in hard-to-understand areas.
- [X] I have run end-to-end tests tests and provided workload links above if applicable.
- [X] I have made or will make corresponding changes to the doc if needed, including adding new documentation pages to the relevant Table of Contents (toctree directive) as explained in [our documentation](https://maxtext.readthedocs.io/en/latest/development.html#adding-new-documentation-files).

### CODE REVIEW

File: src/maxtext/kernels/megablox/ops.py [hunk 1]
```diff
@@ -212,14 +214,12 @@ def _gmm_bwd(
         quantization_rule.bwd_qtype,
         channelwise_axes=[] if quantization_rule.disable_channelwise_axes else [0],
         calibration_method=quantization_rule.bwd_calibration_method,
-        scale_dtype=jnp.float32,
     )
```
Comment (suexu1025) [inline src/maxtext/kernels/megablox/ops.py:217]: do we change scale_dtype here?

Comment (BirdsOfAFthr) [reply src/maxtext/kernels/megablox/ops.py:217]: removing this bc we can leverage bf16 scales, which will use native vpu ops
